### PR TITLE
Room reservations

### PR DESCRIPTION
**Pull request by calebsander**
### Summary
- Added room reservations

### Test Plan
- Try booking some rooms; make sure conflicts are checked correctly
- Make sure they show up under "my reservations"
- Execute some test queries under "all reservations"
- Make sure "room info" is displaying correctly

### CODE REVIEW

File: donut/modules/rooms/templates/mine-iframe.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+					{% for reservation in reservations["upcoming"] %}
+						<div class="list-group-item">
+							<p class='list-group-item-text'>
+								{{ reservation["room"] }} on
+								{{ reservation["start"].strftime("%-m/%-d/%y") }} from
+								{{ reservation["start"].strftime("%-I:%M %p") }} to
+								{{ reservation["end"].strftime("%-I:%M %p") }}
+
+								<a href="/reservation/{{ reservation['id'] }}">View</a>
+							</p>
```
Comment (RobertEng) [inline donut/modules/rooms/templates/mine-iframe.html]: Probably want to allow people to delete their reservations here.

File: donut/modules/rooms/helpers.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        VALUES (:room, :user, :reason, :start, :end)
+    """)
+    flask.g.db.execute(
+        insertion,
+        room=room,
+        user=get_user_id(username),
+        reason=reason,
+        start=start,
+        end=end)
+
```
Comment (RobertEng) [inline donut/modules/rooms/helpers.py:44]: if its not too difficult, return success boolean. or do error checking of some kind.

File: donut/modules/rooms/helpers.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    query = sqlalchemy.text("""
+        DELETE FROM room_reservations
+        WHERE id = :id
+        AND user_id IN (
+            SELECT user_id FROM users WHERE username = :username
+        )
+    """)
+    flask.g.db.execute(query, id=id, username=username)
+
```
Comment (RobertEng) [inline donut/modules/rooms/helpers.py:148]: again if its not too difficult, return success boolean

File: donut/modules/rooms/routes.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        start_hour=None,
+        start_minute=None,
+        end_hour=None,
+        end_minute=None)
+
+
+@blueprint.route("/1/book-room/", methods=["POST"])
+def book():
+    """POST /1/book-room/"""
+
```
Comment (RobertEng) [inline donut/modules/rooms/routes.py]: should be POST request to "/1/reservation"

File: donut/modules/rooms/routes.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        reservations = helpers.get_all_reservations([], now, next_week)
+    return flask.render_template(
+        "all-iframe.html",
+        rooms=helpers.get_rooms(),
+        start=start,
+        end=end,
+        reservations=reservations)
+
+
+@blueprint.route("/reservation/<int:id>", methods=["GET"])
```
Comment (RobertEng) [inline donut/modules/rooms/routes.py]: /1/

File: donut/modules/rooms/routes.py [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+def view_reservation(id):
+    return flask.render_template(
+        "reservation-view.html",
+        reservation=helpers.get_reservation(id),
+        now=datetime.now())
+
+
+@blueprint.route("/reservation/<int:id>", methods=["DELETE"])
```
Comment (RobertEng) [inline donut/modules/rooms/routes.py]: /1/

File: sql/rooms.sql [hunk 7]
```diff
@@ -0,0 +1,25 @@
+DROP TABLE IF EXISTS rooms;
+DROP TABLE IF EXISTS room_reservations;
+
+CREATE TABLE rooms (
+	id INT NOT NULL AUTO_INCREMENT,
```
Comment (RobertEng) [inline sql/rooms.sql]: rename this room_id

File: sql/rooms.sql [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+	location VARCHAR(50) NOT NULL, -- e.g. SAC 23
+	title VARCHAR(50) NOT NULL, -- e.g. ASCIT Screening Room
+	description VARCHAR(255) NOT NULL, -- e.g. A room for watching DVDs and videos
+
+	PRIMARY KEY(id),
+	UNIQUE KEY(location)
+);
+
+CREATE TABLE room_reservations(
+	id INT NOT NULL AUTO_INCREMENT,
```
Comment (RobertEng) [inline sql/rooms.sql]: rename this reservation_id

**Review by RobertEng — CHANGES_REQUESTED** [review]

**Comment by calebsander** [comment]
Alright, I addressed all the issues except for adding success information because I don't want to suppress an error report

**Comment by RobertEng** [comment]
lgtm

**Review by RobertEng — APPROVED** [review]

**Pull request merged**